sfge/src/main/java/com/salesforce/rules/avoiddatabaseoperationinloop/AvoidDatabaseOperationInLoopHandler.java

```diff
@@ -8,35 +8,26 @@
 import com.salesforce.graph.vertex.*;
 import com.salesforce.graph.visitor.ApexPathWalker;
 import com.salesforce.rules.Violation;
-import java.util.Optional;
 import java.util.Set;
 import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
 
 public class AvoidDatabaseOperationInLoopHandler {
 
     /** checks if a certain vertex is of interest to this rule */
     public boolean test(BaseSFVertex vertex) {
-        /* Three different types of DML statements we need to check for:
-           1. insert a;
-           2. [SELECT a FROM b WHERE c]
-           3. Database.something();
-        */
-        if (vertex instanceof DmlStatementVertex) {
-            return true;
-        } else if (vertex instanceof SoqlExpressionVertex) {
-            return true;
-        } else if (vertex instanceof MethodCallExpressionVertex) {
-            MethodCallExpressionVertex methodVertex = (MethodCallExpressionVertex) vertex;
-            final String fullMethodName = methodVertex.getFullMethodName();
+        boolean interested = DatabaseOperationUtil.isDatabaseOperation(vertex);
 
-            // we know this is a method call/expression vertex, but we need to see
-            // if it is a Database.<method> call to confirm it is a DML operation
-            // if so, confirm it should be a violation in a loop, and visit
-            Optional<DmlUtil.DatabaseOperation> operation =
-                    DmlUtil.DatabaseOperation.fromString(fullMethodName);
-            return (operation.isPresent() && operation.get().isViolationInLoop());
+        if (interested
```

Comment:
Since we have encapsulated the logic to see if this vertex is a database operation: if we determine that this is a `MethodCallExpressionVertex` is a database operation (i.e. Database.whatever), we still want to check that it is a violation within a loop, because not all methods in that class are.